spec/models/queries/projects/project_query_results_life_cycle_gate_spec.rb

```diff
@@ -0,0 +1,446 @@
+      end
+
+      it "returns no project as the stage ended in the week before" do
+        Timecop.travel(Date.parse("2025-03-03").noon) do
+          expect(instance.results).to be_empty
+        end
+      end
+    end
+
+    context "when being on Sunday, the gate is on Monday and the week is configured to start on Monday",
+            with_settings: { start_of_week: 1 } do
+      before do
+        gate.update_column(:start_date, Date.parse("2025-03-03"))
+      end
+
+      it "returns the project whose gate is in the current week" do
+        Timecop.travel(Date.parse("2025-03-09").noon) do
+          expect(instance.results).to contain_exactly(project_with_gate)
+        end
+      end
+    end
+
+    context "when being on Sunday, the gate is on Monday and the week is configured to start on Sunday",
+            with_settings: { start_of_week: 7 } do
+      before do
+        gate.update_column(:start_date, Date.parse("2025-03-03"))
+      end
+
+      it "returns no project as the stage ended in the week before" do
+        Timecop.travel(Date.parse("2025-03-09").noon) do
+          expect(instance.results).to be_empty
+        end
+      end
+    end
+  end
+
+  context "with a <>d (between) operator" do
+    before do
+      instance.where(filter_key, "<>d", values)
+    end
+
+    context "when encompassing the gate completely" do
+      let(:values) { [(gate_date - 1.day).to_s, (gate_date + 1.day).to_s] }
+
+      it "returns the project with the gate" do
+        expect(instance.results).to contain_exactly(project_with_gate)
+      end
+    end
+
+    context "when encompassing the gate precisely" do
+      let(:values) { [gate_date.to_s, gate_date.to_s] }
+
+      it "returns the project with the gate" do
+        expect(instance.results).to contain_exactly(project_with_gate)
+      end
+    end
+
+    context "when only the lower value is provided and that one is before the gate's date" do
+      let(:values) { [(gate_date - 1.day).to_s, ""] }
+
+      it "returns the project with the gate" do
+        expect(instance.results).to contain_exactly(project_with_gate)
+      end
+    end
+
+    context "when only the upper value is provided and that one is after the gate's date" do
+      let(:values) { ["", (gate_date + 1.day).to_s] }
+
+      it "returns the project with the gate" do
+        expect(instance.results).to contain_exactly(project_with_gate)
+      end
+    end
+
+    context "when only the upper value is provided and that one is on the gate's date" do
+      let(:values) { ["", gate_date.to_s] }
+
+      it "returns the project with the gate" do
+        expect(instance.results).to contain_exactly(project_with_gate)
+      end
+    end
+
+    context "when only the upper value is provided and that one is before the gate's date" do
+      let(:values) { ["", (gate_date - 1.day).to_s] }
+
+      it "returns no project" do
+        expect(instance.results).to be_empty
+      end
+    end
```

Comment:
The opposite cases for the lower value seem to be missing.

Edit: since the stage spec has the 3 lower cases and 1 upper case, I assume this is intentional.